x-pack/test/fleet_cypress/artifact_manager.ts

```diff
@@ -7,8 +7,14 @@
 
 import axios from 'axios';
 import { last } from 'lodash';
+import pRetry from 'p-retry';
+
+const DEFAULT_VERSION = '8.15.0-SNAPSHOT';
 
 export async function getLatestVersion(): Promise<string> {
-  const response: any = await axios('https://artifacts-api.elastic.co/v1/versions');
-  return last(response.data.versions as string[]) || '8.1.0-SNAPSHOT';
+  return pRetry(() => axios('https://artifacts-api.elastic.co/v1/versions'), {
```

Comment:
The artifacts-api is not stable, don't use it. We had to move away from it in the agent team.

The latest snapshot for each version is available from https://snapshots.elastic.co which does not use the artifacts API.

Examples:
* https://snapshots.elastic.co/latest/8.15.0-SNAPSHOT.json
* https://snapshots.elastic.co/latest/8.14.2-SNAPSHOT.json

The version to use will be the current version of Kibana for the next release. 

Those will give you a build ID you can use to build the artifact download URL For example 8.15.0-SNAPSHOT gives you `"build_id" : "8.15.0-72200925"` as the latest build ID which gets used as shown below to get the linux x64_64 .tar.gz artifact directly.

https://snapshots.elastic.co/8.15.0-72200925/downloads/beats/elastic-agent/elastic-agent-8.15.0-SNAPSHOT-linux-x86_64.tar.gz

Reply:
@cmacknz Looks like a bunch of Kibana tests use the artifacts API, do you think even with retry? (and a fallback value) it's not good enough?  
I am worried when using `https://snapshots.elastic.co/` we will have issues when creating new branch for new version and when the elastic-agent SNAPSHOT is not yet available

Reply:
> I am worried when using https://snapshots.elastic.co/ we will have issues when creating new branch for new version and when the elastic-agent SNAPSHOT is not yet available

Then we might have to pin the exact version indeed. Retrying might help but stability is not guaranteed.

Reply:
> Then we might have to pin the exact version indeed. Retrying might help but stability is not guaranteed.

The way I implemented this is with retry and a fallback value to `8.15.0-SNAPSHOT` if retries failed

Reply:
You can certainly start with a retry and see how it works, if it continues to be flaky the solution is as described above - don't use the artifacts-api.